src/websocket/common.cc

```diff
@@ -60,12 +60,9 @@ future<> connection::send_data(opcodes opcode, temporary_buffer<char>&& buff) {
         header[1] = uint8_t(buff.size());
     }
 
-    scattered_message<char> msg;
-    msg.append(sstring(header, header_size));
-    msg.append(std::move(buff));
-    return _write_buf.write(std::move(msg)).then([this] {
-        return _write_buf.flush();
-    });
+    co_await _write_buf.write(header, header_size);
+    co_await _write_buf.write(std::move(buff));
+    co_await _write_buf.flush();
```

Comment:
The coroutine adds an allocation, but removal of scattered_message removes an allocation, so net zero. In general it's better not to introduce coroutines for functions that are expected not to yield.